Part as planned 1.0.1

jacewski-bosch opened the pull request:
## Description
Improved descriptions of some elements to make versions of part as planned more explicit

Closes #60 

<!-- The MS2 and MS3 criteria are intended for merges to the main-branch. For small bug-fixes or during the model development, for instance, when merging to a feature branch, you may decide to not fill out the checklists. However, we recommend to follow the MS2 checklist during the development. The MS3 checklist becomes relevant for merges to the main-branch. -->
## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the BAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar bamm-cli.jar -i \<path-to-aspect-model\> -v ). The  BAMM CLI is available [here](https://openmanufacturingplatform.github.io/sds-documentation/sds-developer-guide/dev-snapshot/tooling-guide/bamm-cli.html) and in [GitHub](https://github.com/OpenManufacturingPlatform/sds-sdk/releases)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "name" and "description"** in English language. 
- [x] **no duplicate names or preferredNames** within an Aspect (e.g. a Property and the referenced Characteristic should not have the same name)
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the BAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://openmanufacturingplatform.github.io/sds-documentation/bamm-specification/v1.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'

jacewski-bosch committed:
Create metadata.json

jacewski-bosch committed:
Create PartAsPlanned.ttl

updated descriptions to make version of a part more explicit.

jacewski-bosch committed:
Update RELEASE_NOTES.md

jacewski-bosch commented:
Modeling_Team
Model_Update

ZazraltMagic commented:
casing:
-not camel case in :UUIDv4 & :UUIDv4RegularExpression (UUIDv4 -> Uuidv4)

naming:
-characteristic missing in :Timestamp :UUIDv4
-enumeration in :ClassificationCharacteristic

prefName:
-missing prefName in :Timestamp
-prefName: Id, ID, Identifier
-prefName: title case? (Name at Manufacturer, Valid To, Valid From)

human-readable prefName:
-typo in prefName :PartAsPlanned ('Part AsPlanned' -> 'Part As Planned')
-camel case in :ValidityPeriodEntity ('ValidityPeriodEntity' -> 'Validity Period Entity')

description:
-missing description in :ValidityPeriodEntity
-missing description in :ValidityPeriodCharacteristic (the same)

example value:
-missing in :partTypeInformation, :validityPeriod, :validTo, :validFrom

change notes missing in release notes

jacewski-bosch committed:
update PartAsPlanned.ttl

- added changes to release notes
- all characteristics, entities, and constraints now have proper names, preferred names and descriptions
- validFrom and validTo properties now have more meaningful example values.
- fixed some typos in preferred names and descriptions.

jacewski-bosch commented:
> casing: -not camel case in :UUIDv4 & :UUIDv4RegularExpression (UUIDv4 -> Uuidv4)

fixed

> naming: -characteristic missing in :Timestamp :UUIDv4 -enumeration in :ClassificationCharacteristic

fixed, timestamp is now a reference to meta model

> prefName: -missing prefName in :Timestamp -prefName: Id, ID, Identifier -prefName: title case? (Name at Manufacturer, Valid To, Valid From)

fixed consistently to ID, improved title casing

> human-readable prefName: -typo in prefName :PartAsPlanned ('Part AsPlanned' -> 'Part As Planned') -camel case in :ValidityPeriodEntity ('ValidityPeriodEntity' -> 'Validity Period Entity')
> 

fixed

> description: -missing description in :ValidityPeriodEntity -missing description in :ValidityPeriodCharacteristic (the same)

fixed
> 
> example value: -missing in :partTypeInformation, :validityPeriod, :validTo, :validFrom

fixed for simple types
> 
> change notes missing in release notes

added more detail to release notes

ZazraltMagic commented:
@jacewski-bosch looking good.
@bs-jokri can you please approve MS2. Thanks Zaz

bs-jokri reviewed (approved):
@ZazraltMagic did the review and I confirm

bs-jokri commented:
MS3 approval given. Model will be merged as soon as ECA check issues are resolved

jacewski-bosch committed:
Update PartAsPlanned.ttl

updated copyright header years

jacewski-bosch commented:
@bs-jokri please merge (fork merge)

bs-jokri commented:
superseded by PR #122
